orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -233,6 +233,38 @@ def subscriptions_filterable(
     return [{**s.__dict__, "metadata": md} for s, md in sequence]
 
 
+@router.get("/search", response_model=list[SubscriptionWithMetadata])
+def subscriptions_search(
+    response: Response, query: str, range: Optional[str] = None, sort: Optional[str] = None
+) -> list[dict]:
+    """Get subscriptions filtered based on a search query string.
+
+    Args:
+        response: Fastapi Response object
+        query: The search query
+        range: Range
+        sort: Sort
+
+    Returns:
+        List of subscriptions
+
+    """
+    range_ = list(map(int, range.split(","))) if range else None
+    sort_ = sort.split(",") if sort else None
+    logger.info("subscriptions_search() called", range=range_, sort=sort_)
+    stmt = select(SubscriptionTable, SubscriptionMetadataTable.metadata_).join_from(
+        SubscriptionTable, SubscriptionMetadataTable, isouter=True
+    )
+
+    stmt = stmt.join(SubscriptionTable.product).options(
+        contains_eager(SubscriptionTable.product), defer(SubscriptionTable.product_id)
+    )
+    stmt = add_subscription_search_query_filter(stmt, query)
```

Comment:
Will there be a follow up to add this to the graphql?

Reply:
Yes. As part of #288 which we will pick up